wireapp/wire-android
# feat: app lock UI and UX adjustments [WPB-4695]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4695" title="WPB-4695" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4695</a>  [Android] Toggle app lock from the app settings (Password only)
  </td></table>
  <br />
 














----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

- `ConversationOptionSwitch` extracted and renamed to `SettingsOptionSwitch` for easier reusability
- added new type of `SettingsItem` to allow adding also rows that contain switch
- moved app lock options from PrivacySettings to global Settings under “App Settings” group
- added `TurnAppLockOffDialog` for the user  to confirm that he/she wants to turn off the app lock
- implemented logic to show which particular passlock requirements are missing
- adjusted UI according to current designs of “set app lock” and “enter app lock”

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/2134

### Testing

#### How to Test

Open settings and set the app lock, kill the app and open again to see enter the app lock screen.

### Attachments (Optional)

| Before | After&nbsp;&nbsp;&nbsp; |
| ----------- | ------------ |
| ![settings_before](https://github.com/wireapp/wire-android-reloaded/assets/30429749/c329340d-478a-442e-a4ed-b4b6915de44a) | ![settings_after](https://github.com/wireapp/wire-android-reloaded/assets/30429749/1c7330dd-7625-4126-93c2-9cf28832342d) |
| ![set_passcode_before](https://github.com/wireapp/wire-android-reloaded/assets/30429749/9c661254-b1c0-4fbb-b33d-38e9f905d34b) | ![set_passcode_after](https://github.com/wireapp/wire-android-reloaded/assets/30429749/06e4593e-556b-41e0-8e8a-bce65f01107a) |
| ![enter_passcode_before](https://github.com/wireapp/wire-android-reloaded/assets/30429749/fbe47d38-0806-44f4-adeb-0c202e666f15) | ![enter_passcode_after](https://github.com/wireapp/wire-android-reloaded/assets/30429749/1959d79f-096c-4dbc-b3e0-eff8cbcb2160) |

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
